rs/bitcoin/kyt/src/state.rs

```diff
@@ -45,10 +46,48 @@ pub enum FetchTxStatus {
 /// input address will be computed and filled in.
 #[derive(Clone, Debug)]
 pub struct FetchedTx {
-    pub tx: Transaction,
+    pub tx: TransactionInputOutput,
     pub input_addresses: Vec<Option<Address>>,
 }
 
+/// Instead of storing the full Transaction data, we only
+/// store relevant bits, including inputs (which are previous
+/// outputs) and output addresses.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TransactionInputOutput {
+    pub inputs: Vec<PreviousOutput>,
```

Comment:
nit: why not a BTreeSet? This would guarantee that there are no duplicates and also ensure a consistent ordering (e.g. for debug/display representations like on the dashboard)